troposphere/logs.py

```diff
@@ -1,6 +1,21 @@
 from . import AWSObject, AWSProperty
 from .constants import LOGS_ALLOWED_RETENTION_DAYS as RETENTION_DAYS
-from .validators import integer_list_item
+from .validators import integer_list_item, json_checker
+
+
+def validate_resource_policy(policy_document):
```

Comment:
This validation should take into account the other policy document types coming from awacs. This is imported via https://github.com/cloudtools/troposphere/blob/main/troposphere/compat.py
Note: the length checks won't work with some of these awacs types and additional checking needs to be done around the json_checker return types.

Reply:
hi @markpeek So the plan will be

1. build a more general policy document validator
2. cover the types of dict, Policy, PolicyDocumen, and string
3. add tests cover all above

Any other places need to be considered here?

Reply:
That looks like the right list of items. Thanks.